lib/logstash/inputs/github.rb

```diff
@@ -72,6 +72,9 @@ def verify_signature(event,body)
         event.tag("_Invalid_Github_Message")
         is_valid = false
       end
+    elsif @secret_token && !sign_header
```

Comment:
While your change is functionally correct and minimal, there are a couple of anti-patterns within the existing method definition (e.g., `if not`, unnecessary `is_valid` variable) and with this change the reasoning behind this code becomes a little less clear, a little harder to follow, and a little more repetitive.

I'd prefer a functionally-equivalent refactor to this whole method, which I think is an improvement:

~~~ ruby
  def verify_signature(event,body)
    # skip validation if we have no secret token
    return true unless @secret_token

    sign_header = event.get("[headers][x-hub-signature]")
    if sign_header
      hash = 'sha1=' + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha1'), @secret_token, body)
      event.set("hash", hash)
      return true if Rack::Utils.secure_compare(hash, sign_header)
    end

    event.tag("_Invalid_Github_Message")
    return false
  end
~~~

Reply:
Makes sense to me! 7d54a10